### PR TITLE
Terra Gloria (episode 16.2) - walkthrough conversion

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: 

* **Server Mode**: renewal

* **Description of Pull Request**: 

Basic and unfinished version of Terra Gloria (episode 16.2).
## Unfinished
- ~~[Ericsson's daily quest](https://github.com/rathena/rathena/pull/3990/commits/a8289d1b14fdcb1cb03cdf6f7f582234ede3f3bf#diff-2fb732394d5b60394655840886ae7625R8908)~~
- ~~[Milo River's gathering quest](https://github.com/rathena/rathena/pull/3990/commits/a8289d1b14fdcb1cb03cdf6f7f582234ede3f3bf#diff-2fb732394d5b60394655840886ae7625R9590)~~
- ~~[Levedeva's 16022 daily quest](https://github.com/rathena/rathena/pull/3990/commits/a8289d1b14fdcb1cb03cdf6f7f582234ede3f3bf#diff-2fb732394d5b60394655840886ae7625R13324)~~
- ~~[Strasse (exchange npc)](https://github.com/rathena/rathena/pull/3990/commits/a8289d1b14fdcb1cb03cdf6f7f582234ede3f3bf#diff-2fb732394d5b60394655840886ae7625R10798)~~
- ~~[Monsters DB](https://github.com/rathena/rathena/pull/3990/commits/a8289d1b14fdcb1cb03cdf6f7f582234ede3f3bf#diff-16c3460662e574e31eda83721315203dR2784)~~
- [Monsters spawn timer](https://github.com/rathena/rathena/pull/3990/commits/a8289d1b14fdcb1cb03cdf6f7f582234ede3f3bf#diff-af8baffe938fb1688b0d662feb74077bR13)
- ~~[Quests name (quest_db)](https://github.com/rathena/rathena/pull/3990/commits/a8289d1b14fdcb1cb03cdf6f7f582234ede3f3bf#diff-43df541b60840c825f4f8997d7e6fad0R1019)~~

## Note
To avoid exploit there are currently 2 instances of Werner Laboratory central room (officially one), the warp entry being the same. The first (main quest) is a solo instance and the 2nd (daily) a party instance. The instance is a basic version, KRO added Seyren boss fight in the next updates.

Thanks to @admkakaroto  and https://www.divine-pride.net/ !

### CODE REVIEW

**Comment by attackjom** [comment]
found a problem when talk to spica in our private room and leave the room before talk to Nihil, then those NPC will not appear again and quest will stuck in this.

File: npc/re/quests/quests_16_2.txt [hunk 1]
```diff
@@ -0,0 +1,13984 @@
...
+	}
+	mes "You passed through the hole where one person can barely enter.";
+	close2;
+	warp "einbech",44,207;
+	end;
+
+OnTouch:
+	if (terra_gloria_main == 10 || terra_gloria_main == 11) {
+		classchange( 4_CRACK, "Very small hole#16nego0", bc_self );
+		classchange( 4_M_LAZY, "Lazy#16nego06",
```
Comment (attackjom) [inline npc/re/quests/quests_16_2.txt]: Should it " Special Force".

Comment (attackjom) [inline npc/re/quests/quests_16_2.txt]: It should be "Special Force".

Comment (attackjom) [inline npc/re/quests/quests_16_2.txt]:
```suggestion
		mes "Anyone else...? Oh, the next one is the good-hearted ^0000cdold women, I mean ladies^000000. Then you meet the Special Force commander ^0000cdErst^000000...";
```

File: db/re/instance_db.txt [hunk 2]
```diff
@@ -38,3 +38,8 @@
 31,Ritual of Blessing,3600,300,2@mir,101,12
 32,Room of Consciousness,3600,300,1@mir,101,10
 // 33,Sky Fortress Invasion,3600,300,1@sthb,54,67,1@sthc,1@sthd
+34,Heart Hunter War Base 1,3600,300,1@swat,35,65
```
Comment (attackjom) [inline db/re/instance_db.txt]:
```suggestion
34,Heart Hunter War Base 1,3600,300,1@swat,35,64
```

**Comment by attackjom** [comment]
quest_16_2.txt

line 5298 : shoud use "and " not "amd"

line 6201 : should be "Special Force" not "Special_Force"

**Comment by zackdreaver** [comment]
@Atemo 
Does iRO's quest2display for these quests has been translated?

**Comment by Atemo** [comment]
The quest entries are missing on IRO client

**Comment by zackdreaver** [comment]
I'm working on it, around 50% completed, I'll resume the works tomorrow.

**Comment by zackdreaver** [comment]
Do quest ID from 7763 - 7769 part of this episode quest?
It's like event quest to me.

**Comment by Atemo** [comment]
You're right, these quests are not a part of the quest, sorry.

**Comment by zackdreaver** [comment]
I haven't tested this on live server, but it should be good
https://github.com/zackdreaver/ROenglishRE/commit/75ffc4f27b91964677befcdc00dd141a6f75db00

**Comment by Badarosk0** [comment]
From what I've checked, it seems that there is no recognition that the player has completed the Banquet of Heroes quests to start the Terra Gloria quests. It does not start.
@Atemo

**Comment by Atemo** [comment]
>From what I've checked, it seems that there is no recognition that the player has completed the Banquet of Heroes quests to start the Terra Gloria quests. It does not start.

What do you mean? You found an issue in this PR?
It seems on official server that the Terra quests can be started when the character has access to the room of consciousness instance (daily quest). There are several check in this PR for that :
```
	if (banquet_main_quest < 26)	// pre-req : access to room of consciousness (first time)
		end;
```

**Comment by Badarosk0** [comment]
Yes, I realised. Even completing the instance the variable did not receive the value. I needed to change the value of the variable to (26) to continue.

**Comment by Atemo** [comment]
I see, the check should be `if (banquet_main_quest < 25)`.

**Comment by admkakaroto** [comment]
A small contribution.
Here: https://github.com/admkakaroto/rathena/commit/1a19c35fa9b9d4658f56d56b9c3f67784d4efffd

**Comment by Badarosk0** [comment]
Map server crash: @Atemo 
```
Program received signal SIGSEGV, Segmentation fault.
map_setcell (m=Unhandled dwarf expression opcode 0xf3
) at map.cpp:3312
3312			case CELL_SHOOTABLE:     mapdata->cell[j].shootable = flag;     break;
Missing separate debuginfos, use: debuginfo-install glibc-2.12-1.209.el6_9.2.x86_64 keyutils-libs-1.4-5.el6.x86_64 krb5-libs-1.10.3-65.el6.x86_64 libcom_err-1.41.12-23.el6.x86_64 libgcc-4.4.7-18.el6.x86_64 libselinux-2.0.94-7.el6.x86_64 libstdc++-4.4.7-18.el6.x86_64 mysql-libs-5.1.73-8.el6_8.x86_64 nss-softokn-freebl-3.14.3-23.3.el6_8.x86_64 openssl-1.0.1e-57.el6.x86_64 pcre-7.8-7.el6.x86_64 zlib-1.2.3-29.el6.x86_64
(gdb) 
[K(gdb) bt full
#0  map_setcell (m=Unhandled dwarf expression opcode 0xf3
) at map.cpp:3312
        j = 6116
#1  0x0000000000530c54 in map_iwall_remove (wall_name=Unhandled dwarf expression opcode 0xf3
) at map.cpp:3450
        iwall = 0x7fffd75b9dac
        i = 0
        x1 = 116
        y1 = 30
#2  0x00000000005c6737 in buildin_delwall (st=0x7fffedd6b7e4) at script.cpp:14115
        name = <value optimized out>
#3  0x00000000005c346d in run_func (st=0x7fffedd6b7e4) at script.cpp:4022
        data = <value optimized out>
        i = <value optimized out>
        start_sp = <value optimized out>
        end_sp = <value optimized out>
        func = 20892
#4  0x00000000005c84f0 in run_script_main (st=0x7fffedd6b7e4) at script.cpp:4297
        cmdcount = 655071
        gotocount = 2048
        sd = <value optimized out>
        stack = 0x7fffe9f9873c
#5  0x00000000005553f3 in npc_instanceinit (nd=0x7fffd4b9d10c) at npc.cpp:3469
        ev = 0x5924bf4
        evname = "dup_2_110020417::OnInstanceInit\000p\345n\000\000\000\000\000\000\000\000\000\245\003\000\000@.\237\000\000\000\000\000\374\276\265\331\377\177\000\000`n\370\000\000\000\000\000\000\000\000\000)"
#6  0x0000000000529974 in map_foreachinareaV(int (*)(block_list *, typedef __va_list_tag __va_list_tag *), int16, int16, int16, int16, int16, int, typedef __va_list_tag __va_list_tag *, bool) (func=0x513f70 <instance_npcinit(block_list *, typedef __va_list_tag __va_list_tag *)>, m=1065, x0=<value optimized out>, 
    y0=<value optimized out>, x1=<value optimized out>, y1=<value optimized out>, type=128, ap=<value optimized out>, wall_check=false) at map.cpp:804
        bx = <value optimized out>
        by = <value optimized out>
        cx = 0
        cy = 64
        returnCount = 0
        bl = <value optimized out>
        blockcount = <value optimized out>
        i = 82
        ap_copy = void
#7  0x0000000000529b63 in map_foreachinallarea(int (*)(block_list *, typedef __va_list_tag __va_list_tag *), int16, int16, int16, int16, int16, int, ...) (func=Unhandled dwarf expression opcode 0xf3
)
    at map.cpp:820
        returnCount = 0
        ap = void
#8  0x0000000000514473 in instance_addnpc (im=0x9f2e40) at instance.cpp:358
        mapdata = <value optimized out>
        i = 0
#9  0x0000000000514687 in instance_addmap (instance_id=2) at instance.cpp:527
        i = <value optimized out>
        m = <value optimized out>
        im = 0x9f2e40
        db = 0x7fffeaf96744
        entry = <value optimized out>
#10 0x00000000005147cd in instance_addmap (tid=Unhandled dwarf expression opcode 0xf3
) at instance.cpp:115
No locals.
#11 instance_subscription_timer (tid=Unhandled dwarf expression opcode 0xf3
) at instance.cpp:119
        i = <value optimized out>
        ret = <value optimized out>
        instance_id = 2
        sd = 0x0
        pd = 0x0
        gd = 0x0
        cd = 0x0
        mode = <value optimized out>
#12 0x0000000000514bf7 in instance_create (owner_id=51064, name=0x7fffde561c29 "Heart Hunter War Base 2", mode=IM_PARTY) at instance.cpp:456
        db = <value optimized out>
        sd = 0x0
        pd = 0x7fffd63b2964
        gd = 0x0
        cd = 0x0
        i = 2
        __FUNCTION__ = "instance_create"
#13 0x00000000005de3d1 in buildin_instance_create (st=0x7fffedd6bd04) at script.cpp:21210
        mode = <value optimized out>
        owner_id = 51064
        __FUNCTION__ = "buildin_instance_create"
#14 0x00000000005c346d in run_func (st=0x7fffedd6bd04) at script.cpp:4022
        data = <value optimized out>
        i = <value optimized out>
        start_sp = <value optimized out>
        end_sp = <value optimized out>
        func = 20945
#15 0x00000000005c84f0 in run_script_main (st=0x7fffedd6bd04) at script.cpp:4297
        cmdcount = 655339
        gotocount = 2045
        sd = <value optimized out>
        stack = 0x7fffe9f963ec
#16 0x000000000054db29 in npc_scriptcont (sd=0x7c0b170, id=110020088, closing=false) at npc.cpp:1428
        target = 0x7fffde168984
        __FUNCTION__ = "npc_scriptcont"
#17 0x00000000004cb5a8 in clif_parse (fd=Unhandled dwarf expression opcode 0xf3
) at clif.cpp:21779
        cmd = 184
        packet_len = 7
        sd = 0x7c0b170
        pnum = 0
#18 0x00000000006885dd in do_sockets (next=Unhandled dwarf expression opcode 0xf3
) at socket.cpp:1408
        rfd = {fds_bits = {4194304, 0 <repeats 15 times>}}
        timeout = {tv_sec = 0, tv_usec = 10792}
        ret = 0
        i = 22
#19 0x000000000046dda5 in main (argc=1, argv=0x7fffffffde28) at core.cpp:369
        next = <value optimized out>
```

**Comment by Atemo** [comment]
The issue seems related to `delwall` command if you know the step please open a new issue to keep the PR cleaner ;)

**Comment by Badarosk0** [comment]
At some point I created a custom instance using delwall and my map server was always knocked over. This problem occurred whenever several people did the instance at the same time. Whenever I tested on the test server individually this did not happen. I was thinking that I was using the command incorrectly, so I removed wall from my script for the problem to stop occurring. It looks like these script wall with instance are not matched well.

**Comment by Atemo** [comment]
This version doesn't have the script to summon seyren. For now it's not useful to implement it

**Comment by DepTrai92** [comment]
hopefully can have Werner Laboratory central room instance work.
i missing this instance,other all i have it

**Comment by blackatz456** [comment]
maybe HeartHunterWarBase can use script setcell instead of setwall and delwall?

**Comment by ecdarreola** [comment]
questinfo requirement update (#4168) have many conflicts for this PR.

**Comment by teededung** [comment]
Monster `Elena Bolkova` (3630) is wrong stats.
Corrected:
`3630,EP16_2_E_BOLKOBA,Elena Bolkova,Elena Bolkova,115,2000000,100,300000,150000,7,1200,300,120,70,80,120,66,50,250,60,10,12,1,7,64,0x3885,150,864,1268,480,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0`

Skills:

```
3630,Elena Bolkova@NPC_STUNATTACK,attack,179,1,1000,500,5000,no,target,always,,,,,,,,
3630,Elena Bolkova@NPC_PULSESTRIKE,attack,661,1,500,2000,5000,no,target,always,,,,,,,,0
3630,Elena Bolkova@NPC_WIDESIGHT,attack,669,1,10000,0,30000,yes,self,always,,,,,,,,
3630,Elena Bolkova@NPC_MAGICMIRROR,attack,671,10,2000,3000,20000,no,self,myhpltmaxrate,20,,,,,,39,
3630,Elena Bolkova@NPC_CRITICALWOUND,attack,673,5,500,1000,30000,no,self,always,,,,,,,,
3630,Elena Bolkova@NPC_AGIUP,chase,350,1,10000,0,20000,yes,self,always,,,,,,,,
```

Information from divine-pride.net

**Comment by admkakaroto** [comment]
Crash on map_server, dellwall issue:

```
Program terminated with signal 11, Segmentation fault.
#0  0x00000000004d846b in map_setcell (m=<optimized out>, x=116, y=<optimized out>,
    cell=cell@entry=CELL_SHOOTABLE, flag=flag@entry=true) at map.cpp:3316
3316                    case CELL_SHOOTABLE:     mapdata->cell[j].shootable = flag;     break;
Missing separate debuginfos, use: debuginfo-install glibc-2.17-260.el7_6.6.x86_64 keyutils-libs-1.5.8-3.el7.x86_64 krb5-libs-1.15.1-37.el7_6.x86_64 libcom_err-1.42.9-13.el7.x86_64 libgcc-4.8.5-36.el7_6.2.x86_64 libselinux-2.5-14.1.el7.x86_64 libstdc++-4.8.5-36.el7_6.2.x86_64 mariadb-libs-5.5.60-1.el7_5.x86_64 openssl-libs-1.0.2k-16.el7_6.1.x86_64 pcre-8.32-17.el7.x86_64 zlib-1.2.7-18.el7.x86_64
(gdb) bt full
#0  0x00000000004d846b in map_setcell (m=<optimized out>, x=116, y=<optimized out>,
    cell=cell@entry=CELL_SHOOTABLE, flag=flag@entry=true) at map.cpp:3316
        j = 6116
#1  0x00000000004d8b96 in map_iwall_remove (wall_name=<optimized out>) at map.cpp:3454
        iwall = 0x7fba1a9a57bc
        i = 0
        x1 = 116
        y1 = 30
#2  0x000000000056f747 in buildin_delwall (st=0x7fba2a5d21e4) at script.cpp:14273
        name = <optimized out>
#3  0x000000000056c36d in run_func (st=st@entry=0x7fba2a5d21e4) at script.cpp:4057
        data = <optimized out>
        i = <optimized out>
        start_sp = <optimized out>
        func = 1626
        end_sp = <optimized out>
#4  0x0000000000571300 in run_script_main (st=0x7fba2a5d21e4) at script.cpp:4332
        c = C_FUNC
        cmdcount = 655071
        gotocount = 2048
        sd = <optimized out>
        stack = 0x7fba2529f414
#5  0x00000000004fd2da in npc_instanceinit (nd=0x7fba186e2dfc) at npc.cpp:3459
        ev = 0x7fba18266ea4
        evname = "dup_4_110008452::OnInstanceInit\000\330bl\000\000\000\000\000\000\000\000\000#\004\000\000`j\233\000\000\000\000\000\304\373I\026\272\177\000\000\300\331\367\000\000\000\000\000\000\000\000\000T"
#6  0x00000000004d1ca4 in map_foreachinareaV (func=
    0x4b4760 <instance_npcinit(block_list *, typedef __va_list_tag __va_list_tag *)>, m=1108,
    x0=<optimized out>, x0@entry=0, y0=<optimized out>, y0@entry=0, x1=<optimized out>,
    y1=<optimized out>, type=type@entry=128, ap=<optimized out>, ap@entry=0x7ffed82aa4a8,
    wall_check=wall_check@entry=false) at map.cpp:803
        by = <optimized out>
        cx = 110
---Type <return> to continue, or q <return> to quit---
        ap_copy = {{gp_offset = 48, fp_offset = 48, overflow_arg_area = 0x7ffed82aa588,
            reg_save_area = 0x7ffed82aa4c0}}
        cy = 119
        blockcount = 0
        bx = <optimized out>
        returnCount = 0
        bl = <optimized out>
        i = 82
#7  0x00000000004d1e21 in map_foreachinallarea (
    func=func@entry=0x4b4760 <instance_npcinit(block_list *, typedef __va_list_tag __va_list_tag *)>, m=<optimized out>, x0=x0@entry=0, y0=y0@entry=0, x1=<optimized out>, y1=<optimized out>,
    type=type@entry=128) at map.cpp:819
        returnCount = 0
        ap = {{gp_offset = 48, fp_offset = 48, overflow_arg_area = 0x7ffed82aa588,
            reg_save_area = 0x7ffed82aa4c0}}
#8  0x00000000004b4c78 in instance_addnpc (im=im@entry=0x9b6a60 <instance_data+256>)
    at instance.cpp:358
        mapdata = <optimized out>
        i = 0
#9  0x00000000004b4e88 in instance_addmap (instance_id=instance_id@entry=4) at instance.cpp:527
        i = <optimized out>
        m = <optimized out>
        im = 0x9b6a60 <instance_data+256>
        db = 0x7fba26031e9c
        __FUNCTION__ = "instance_addmap"
        entry = <optimized out>
#10 0x00000000004b4f92 in instance_subscription_timer (tid=tid@entry=0, tick=tick@entry=0,
    id=id@entry=0, data=data@entry=0) at instance.cpp:119
        i = <optimized out>
        ret = <optimized out>
        instance_id = 4
        pd = 0x0
        sd = 0x0
        gd = 0x0
---Type <return> to continue, or q <return> to quit---
        cd = 0x0
        mode = IM_PARTY
#11 0x00000000004b5392 in instance_create (owner_id=owner_id@entry=164231,
    name=0x7fba20600ea6 "Heart Hunter War Base 1", mode=mode@entry=IM_PARTY) at instance.cpp:456
        db = <optimized out>
        pd = 0x7fba150c890c
        i = 4
        __FUNCTION__ = "instance_create"
        sd = 0x0
        gd = 0x0
        cd = 0x0
#12 0x0000000000586dec in buildin_instance_create (st=0x7fba2a5d20c4) at script.cpp:21249
        mode = IM_PARTY
        owner_id = 164231
        __FUNCTION__ = "buildin_instance_create"
#13 0x000000000056c36d in run_func (st=st@entry=0x7fba2a5d20c4) at script.cpp:4057
        data = <optimized out>
        i = <optimized out>
        start_sp = <optimized out>
        func = 1678
        end_sp = <optimized out>
#14 0x0000000000571300 in run_script_main (st=0x7fba2a5d20c4) at script.cpp:4332
        c = C_FUNC
        cmdcount = 655339
        gotocount = 2046
        sd = <optimized out>
        stack = 0x7fba2529f384
#15 0x00000000004f5b3b in npc_scriptcont (sd=0x9e87f80, id=110008123, closing=false)
    at npc.cpp:1428
        target = 0x7fba202637dc
        __FUNCTION__ = "npc_scriptcont"
#16 0x000000000046b5a9 in clif_parse (fd=20) at clif.cpp:22287
        cmd = 184
        packet_len = 7
---Type <return> to continue, or q <return> to quit---
        sd = 0x9e87f80
        pnum = 0
#17 0x000000000063016d in do_sockets (next=<optimized out>) at socket.cpp:1407
        rfd = {fds_bits = {1048576, 0 <repeats 15 times>}}
        timeout = {tv_sec = 0, tv_usec = 2345}
        ret = 0
        i = 20
#18 0x000000000040764d in main (argc=1, argv=0x7ffed82aa9c8) at core.cpp:369
        next = <optimized out>
(gdb)
```

**Comment by Badarosk0** [comment]
> Crash on map_server, dellwall issue:
> 
> ```
> Program terminated with signal 11, Segmentation fault.
> #0  0x00000000004d846b in map_setcell (m=<optimized out>, x=116, y=<optimized out>,
>     cell=cell@entry=CELL_SHOOTABLE, flag=flag@entry=true) at map.cpp:3316
> 3316                    case CELL_SHOOTABLE:     mapdata->cell[j].shootable = flag;     break;
> Missing separate debuginfos, use: debuginfo-install glibc-2.17-260.el7_6.6.x86_64 keyutils-libs-1.5.8-3.el7.x86_64 krb5-libs-1.15.1-37.el7_6.x86_64 libcom_err-1.42.9-13.el7.x86_64 libgcc-4.8.5-36.el7_6.2.x86_64 libselinux-2.5-14.1.el7.x86_64 libstdc++-4.8.5-36.el7_6.2.x86_64 mariadb-libs-5.5.60-1.el7_5.x86_64 openssl-libs-1.0.2k-16.el7_6.1.x86_64 pcre-8.32-17.el7.x86_64 zlib-1.2.7-18.el7.x86_64
> (gdb) bt full
> #0  0x00000000004d846b in map_setcell (m=<optimized out>, x=116, y=<optimized out>,
>     cell=cell@entry=CELL_SHOOTABLE, flag=flag@entry=true) at map.cpp:3316
>         j = 6116
> #1  0x00000000004d8b96 in map_iwall_remove (wall_name=<optimized out>) at map.cpp:3454
>         iwall = 0x7fba1a9a57bc
>         i = 0
>         x1 = 116
>         y1 = 30
> #2  0x000000000056f747 in buildin_delwall (st=0x7fba2a5d21e4) at script.cpp:14273
>         name = <optimized out>
> #3  0x000000000056c36d in run_func (st=st@entry=0x7fba2a5d21e4) at script.cpp:4057
>         data = <optimized out>
>         i = <optimized out>
>         start_sp = <optimized out>
>         func = 1626
>         end_sp = <optimized out>
> #4  0x0000000000571300 in run_script_main (st=0x7fba2a5d21e4) at script.cpp:4332
>         c = C_FUNC
>         cmdcount = 655071
>         gotocount = 2048
>         sd = <optimized out>
>         stack = 0x7fba2529f414
> #5  0x00000000004fd2da in npc_instanceinit (nd=0x7fba186e2dfc) at npc.cpp:3459
>         ev = 0x7fba18266ea4
>         evname = "dup_4_110008452::OnInstanceInit\000\330bl\000\000\000\000\000\000\000\000\000#\004\000\000`j\233\000\000\000\000\000\304\373I\026\272\177\000\000\300\331\367\000\000\000\000\000\000\000\000\000T"
> #6  0x00000000004d1ca4 in map_foreachinareaV (func=
>     0x4b4760 <instance_npcinit(block_list *, typedef __va_list_tag __va_list_tag *)>, m=1108,
>     x0=<optimized out>, x0@entry=0, y0=<optimized out>, y0@entry=0, x1=<optimized out>,
>     y1=<optimized out>, type=type@entry=128, ap=<optimized out>, ap@entry=0x7ffed82aa4a8,
>     wall_check=wall_check@entry=false) at map.cpp:803
>         by = <optimized out>
>         cx = 110
> ---Type <return> to continue, or q <return> to quit---
>         ap_copy = {{gp_offset = 48, fp_offset = 48, overflow_arg_area = 0x7ffed82aa588,
>             reg_save_area = 0x7ffed82aa4c0}}
>         cy = 119
>         blockcount = 0
>         bx = <optimized out>
>         returnCount = 0
>         bl = <optimized out>
>         i = 82
> #7  0x00000000004d1e21 in map_foreachinallarea (
>     func=func@entry=0x4b4760 <instance_npcinit(block_list *, typedef __va_list_tag __va_list_tag *)>, m=<optimized out>, x0=x0@entry=0, y0=y0@entry=0, x1=<optimized out>, y1=<optimized out>,
>     type=type@entry=128) at map.cpp:819
>         returnCount = 0
>         ap = {{gp_offset = 48, fp_offset = 48, overflow_arg_area = 0x7ffed82aa588,
>             reg_save_area = 0x7ffed82aa4c0}}
> #8  0x00000000004b4c78 in instance_addnpc (im=im@entry=0x9b6a60 <instance_data+256>)
>     at instance.cpp:358
>         mapdata = <optimized out>
>         i = 0
> #9  0x00000000004b4e88 in instance_addmap (instance_id=instance_id@entry=4) at instance.cpp:527
>         i = <optimized out>
>         m = <optimized out>
>         im = 0x9b6a60 <instance_data+256>
>         db = 0x7fba26031e9c
>         __FUNCTION__ = "instance_addmap"
>         entry = <optimized out>
> #10 0x00000000004b4f92 in instance_subscription_timer (tid=tid@entry=0, tick=tick@entry=0,
>     id=id@entry=0, data=data@entry=0) at instance.cpp:119
>         i = <optimized out>
>         ret = <optimized out>
>         instance_id = 4
>         pd = 0x0
>         sd = 0x0
>         gd = 0x0
> ---Type <return> to continue, or q <return> to quit---
>         cd = 0x0
>         mode = IM_PARTY
> #11 0x00000000004b5392 in instance_create (owner_id=owner_id@entry=164231,
>     name=0x7fba20600ea6 "Heart Hunter War Base 1", mode=mode@entry=IM_PARTY) at instance.cpp:456
>         db = <optimized out>
>         pd = 0x7fba150c890c
>         i = 4
>         __FUNCTION__ = "instance_create"
>         sd = 0x0
>         gd = 0x0
>         cd = 0x0
> #12 0x0000000000586dec in buildin_instance_create (st=0x7fba2a5d20c4) at script.cpp:21249
>         mode = IM_PARTY
>         owner_id = 164231
>         __FUNCTION__ = "buildin_instance_create"
> #13 0x000000000056c36d in run_func (st=st@entry=0x7fba2a5d20c4) at script.cpp:4057
>         data = <optimized out>
>         i = <optimized out>
>         start_sp = <optimized out>
>         func = 1678
>         end_sp = <optimized out>
> #14 0x0000000000571300 in run_script_main (st=0x7fba2a5d20c4) at script.cpp:4332
>         c = C_FUNC
>         cmdcount = 655339
>         gotocount = 2046
>         sd = <optimized out>
>         stack = 0x7fba2529f384
> #15 0x00000000004f5b3b in npc_scriptcont (sd=0x9e87f80, id=110008123, closing=false)
>     at npc.cpp:1428
>         target = 0x7fba202637dc
>         __FUNCTION__ = "npc_scriptcont"
> #16 0x000000000046b5a9 in clif_parse (fd=20) at clif.cpp:22287
>         cmd = 184
>         packet_len = 7
> ---Type <return> to continue, or q <return> to quit---
>         sd = 0x9e87f80
>         pnum = 0
> #17 0x000000000063016d in do_sockets (next=<optimized out>) at socket.cpp:1407
>         rfd = {fds_bits = {1048576, 0 <repeats 15 times>}}
>         timeout = {tv_sec = 0, tv_usec = 2345}
>         ret = 0
>         i = 20
> #18 0x000000000040764d in main (argc=1, argv=0x7ffed82aa9c8) at core.cpp:369
>         next = <optimized out>
> (gdb)
> ```

Yes this also happens to me.

File: npc/re/instances/HeartHunterWarBase.txt [hunk 3]
```diff
@@ -1834,12 +1828,7 @@ OnInstanceInit:
...
 
-	if (checkwall("swat_2_wall_step_1_" + 'instance_id) == true)
-		delwall "swat_2_wall_step_1_" + 'instance_id;
-	if (checkwall("swat_2_wall_step_2_" + 'instance_id) == true)
-		delwall "swat_2_wall_step_2_" + 'instance_id;
-
-	setwall 'map_swat$,23,222, 17, DIR_EAST, false, "swat_2_wall_step_1_" + 'instance_id;
-	setwall 'map_swat$,22,272, 30, DIR_EAST, false, "swat_2_wall_step_2_" + 'instance_id;
+	setcell 'map_swat$,23,222,40,222,cell_walkable,1;
+	setcell 'map_swat$,22,272,52,272,cell_walkable,1;
```
Comment (Angelic234) [inline npc/re/instances/HeartHunterWarBase.txt]: ALL setcell flag ars opposite.

**Comment by RagnaWay** [comment]
Update quest_db please!

**Comment by ecdarreola** [comment]
> Update quest_db please!

Instead of saying update quest_db please, if you're using the latest revision of rathena, just compile your server and run csv2yaml converter. It will automatically convert quest_db.txt to .yml format.

**Comment by admkakaroto** [comment]
Please update.

**Comment by Atemo** [comment]
The Werner Laboratory central room instance is a basic version, it doesn't have the fight against Seyren. I don't have the data for the quest/instance so I let it like that for now.

File: src/map/status.cpp [hunk 4]
```diff
@@ -1393,6 +1393,11 @@ void initChangeTables(void)
 	StatusIconChangeTable[SC_USE_SKILL_SP_SPA] = EFST_USE_SKILL_SP_SPA;
 	StatusIconChangeTable[SC_USE_SKILL_SP_SHA] = EFST_USE_SKILL_SP_SHA;
 
+	// ep16.2
+	StatusIconChangeTable[SC_EP16_2_BUFF_SS] = EFST_EP16_2_BUFF_SS;
+	StatusIconChangeTable[SC_EP16_2_BUFF_SC] = EFST_EP16_2_BUFF_SC;
+	StatusIconChangeTable[SC_EP16_2_BUFF_AC] = EFST_EP16_2_BUFF_AC;
```
[the remark this answers is not in the document]
Comment (Atemo) [reply src/map/status.cpp:1399]: According to my tests the buffs are
```
* Not removed on change map
* Not removed on log out
* Removed on death
```
I don't know if they are removed by clearance but most buffs not removed on death are also not removed by clearance, so I assumed the buffs can be removed by clearance. Need further testing.

**Comment by Atemo** [comment]
Any last review before merging?

File: src/map/skill.cpp [hunk 5]
```diff
@@ -8484,6 +8485,7 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 					case SC_DONTFORGETME:
 					case SC_FORTUNE:
 					case SC_SERVICE4U:
+					case SC_EP16_2_BUFF_SS:		case SC_EP16_2_BUFF_SC:	case SC_EP16_2_BUFF_AC:
```
Comment (aleos89) [inline src/map/skill.cpp:8488]: These need to be moved up to the `continue;` else it will hit the check specific for songs.

**Comment by saya9200** [comment]
currently running this on clean rA 

the instance is stuck. 

grafik


1@swat,163,58,0	script	Pile of gunpowder#2	4_ENERGY_YELLOW,{
	if ('event != 3)
		end;
	'event = 4;
	specialeffect EF_COIN;
	disablenpc instance_npcname("Pile of gunpowder#2");
	initnpctimer;
	end;
OnTimer1000:
	stopnpctimer;
	for ( .@i = 0; .@i < 5; .@i++ )
		makeitem 25154,1,'map_swat$, rand(161,165), rand(51,55);// Antique_Powder
	end;
}


 I was on @speed 1
Killed monsters very fast.

Now clicking the pile of gunpowder does nothing and the instance cant progress.

**Comment by Dieby** [comment]
Update daily mission YSF01 Seyren please quests_16_2

**Comment by Dieby** [comment]
mes "[Rookie]";
	mes "Shall we leave now?";
	cutin "16loo_01.bmp",2;
	next;
	switch( select( "About the monster in the central room", "Leave.", "Stay.", "About YSF01" ) ) {
	case 1:
		break;
	case 2:
		mes "[Rookie]";
		mes "Follow me, then.";
		close2;
		warp "ein_fild04",279,339;
		end;
	case 3:
		mes "[Rookie]";
		mes "Since this fuss is likely to continue for a while, we'd be glad if you would stay here and help us~";
		close3;
	case 4:
		mes "[Rookie]";
		mes "Why did Werner do that?";
		mes "I don't understand.";
		mes "... No. Maybe it's weird to understand a lunatic.";
		next;
		mes "[Rookie]";
		mes "Anyways, We have to observe the central room's core.";
		mes "As long as there are magic inside, the same thing will happen again.";
		close3;
	}


YSF01 Seyren daily
quests_16_2.txt
Line 3489

Update Please

**Comment by Atemo** [comment]
I don't have the data for YSF01 hunting quest.

**Pull request merged**